lib/active_merchant/billing/gateways/barclaycard_smartpay.rb

```diff
@@ -163,7 +188,7 @@ def post_data(data)
 
       def message_from(response)
         return response['resultCode'] if response.has_key?('resultCode') # Payment request
-        return response['response'] if response['response'] # Modification request
+        return response['response'] if response.has_key?('response') # Modification request
```

Comment:
It's not, just noticed it and tidied it up; but in fact it probably works without .has_key? which would be even tidier.  But I'll remove it since it's unrelated to the feature.